docs/dependency-injection/dependency-injection.md

```diff
@@ -0,0 +1,206 @@
+
+In a typical Oryx application, much of the work related to setting up the DI container is handled automatically by the application orchestrator. However, the @spryker-oryx/di library provides a  set of tools that can be used to further customize and fine-tune the DI as needed.
+
+# 3. Using services
+
+## inject vs resolve
+
+There are two primary methods for injecting services and dependencies into components: `inject` and `resolve`.
+
+### inject()
+
+The `inject` method is used to inject services from the current injector. This method can be used when the injection context is defined, such as in services at construction time or in provided factories that use `useFactory`.
+
+For example, consider the following code:
+
+```typescript
+export class MyService { 
+   constructor(protected otherService = inject(SomeOtherService)) {} 
+}
+```
+
+In this example, the `MyService` constructor injects an instance of `SomeOtherService` using the `inject()` utility.
+
+### resolve()
+
+The `resolve` method works similarly to `inject`, but it tries to resolve the current injector using context (by default globalThis). This method is suitable to use in components, especially in web components, where the injection context may not be defined at construction time.
+
+For example, consider the following code:
+
+```typescript
+export class MyComponent extends LitElement { 
+   portected myService = resolve(MyService); 
+}
+```
+
+In this example, the `MyComponent` class uses the `resolve` utility to inject an instance of `MyService` into the component. Since `MyComponent` does not have an explicit injector defined at construction time, `resolve` uses the global context to resolve the injector.
+
+
+## Injecting INJECTOR
+
+
+In some cases, dependencies may be optional or resolved at runtime, making it difficult to inject them directly into a service. In these situations, you can inject the current DI container using the `INJECTOR` token, like in the following example:
+
+
+```typescript
+constructor(protected injector = inject(INJECTOR)) {}
+```
+
+
+With the `INJECTOR` token injected, you can use the injector's `inject` method to dynamically resolve dependencies at runtime. For example:
+
+```typescript
+someServiceMethod(field) {
+	const template = this.injector.inject(`${FormFieldRenderer}-${field.type}`);
+}
+
+```
+
+In this example, the `inject` method is used to resolve a dependency based on a dynamic token that includes the field type. Since the field type is only known at runtime, it is not possible to inject the dependency directly. Instead, the `INJECTOR` token is injected, allowing the `inject` method to resolve the dependency dynamically.
+
+
+# 4. Defining services
+
+In Oryx, we follow certain conventions for defining services.
+
+## String-Based Token
+
+Services are resolved using string tokens. For example:
+
+```typescript
+export const CartService = 'FES.CartService';
```

Comment:
```suggestion
export const CartService = 'oryx.CartService';
```

Reply:
Let's add a recommendation that a project specific prefix should be used to avoid conflicts.